### DISCUSSION TITLE
Can't install a package from VCS repository that requires another package that is also from VCS repository

### DISCUSSION THREAD

**Discussion by ghost** [composer/composer#11857] [Q&A / Support]
Both packages exist on satis and **releases** section contains branch `task-123`.
In my application root I run:
```sh
$ composer require vendor/package-a:dev-task-123 as 1.0
Problem 1:
  - vendor/package-a dev-task-123 requires vendor/package-b task-123 -> no matching package found.
```

Main composer.json:
```json
{
  "repositories": [
    {
      "type": "composer",
      "url": "https://satis.example.com"
    }
  ]
}
```

Package A composer.json:
```json
{
  "repositories": [
    {
      "type": "composer",
      "url": "https://satis.example.com"
    }
  ]
  "require": {
    "vendor/package-b": "dev-task-123 as 1.0"
  }
}
```

**Comment by fredden** [comment]
"repositories" is only read for the root package, so that doesn't seem relevant here. is this perhaps a `minimum-stability` issue? Does this work when set to `dev`?

**Reply by ghost** [reply]
Weird thing is that it works when I require both package-a and package-b in the root composer.json.
